### PR TITLE
Users api

### PR DESCRIPTION
**Pull request by 31453**
Put all users api endpoint functions into apiUsers.js!

Document and standardize endpoints.

### CODE REVIEW

File: viewer/apiUsers.js [hunk 1]
```diff
@@ -0,0 +1,1493 @@
+'use strict';
+
+const fs = require('fs');
+const stylus = require('stylus');
+
+module.exports = (app, Config, Db, internals, ViewerUtils) => {
```
Comment (awick) [inline viewer/apiUsers.js:6]: would like to remove app

File: viewer/apiUsers.js [hunk 2]
```diff
@@ -0,0 +1,1493 @@
...
+
+    const clone = {};
+
+    for (const prop of userProps) {
+      if (req.user.hasOwnProperty(prop)) {
+        clone[prop] = req.user[prop];
+      }
+    }
+
+    clone.canUpload = app.locals.allowUploads;
```
Comment (awick) [inline viewer/apiUsers.js:204]: move to internals

File: viewer/vueapp/src/components/settings/Settings.vue [hunk 3]
```diff
@@ -2499,11 +2499,11 @@ export default {
 
           if (updateTheme) {
             let now = Date.now();
-            if ($('link[href^="user.css"]').length) {
-              $('link[href^="user.css"]').remove();
+            if ($('link[href^="api/user/css"]').length) {
```
Comment (awick) [inline viewer/vueapp/src/components/settings/Settings.vue:2502]: still hurts deeply that it isn't api/user.css, at least we support both

**Review by awick — CHANGES_REQUESTED** [review]

File: viewer/vueapp/src/components/settings/Settings.vue (hunk 3 above)
Comment (31453) [reply viewer/vueapp/src/components/settings/Settings.vue:2502]: 💔

File: viewer/apiUsers.js (hunk 2 above)
Comment (31453) [reply viewer/apiUsers.js]: 👍

File: viewer/apiUsers.js (hunk 1 above)
Comment (31453) [reply viewer/apiUsers.js]: 👍

**Pull request merged**